libtcmu_log.c

```diff
@@ -714,13 +714,17 @@ static bool is_same_path(const char* path1, const char* path2)
 int tcmu_resetup_log_file(char *log_dir)
 {
 	int ret;
+	char old_logdir[PATH_MAX] = {'\0', };
```

Comment:
I think tcmu_log_dir_set should have been called by the tcmu_log_dir_create caller and not tcmu_log_dir_create. tcmu_log_dir_create just creates the dirs. Calling it in tcmu_log_dir_create leads to the weird dance we do below and also if the resetup fails we were leaving it set to the bad value.

Reply:
Yeah, agree with this.